static-site/src/components/splash/index.jsx

```diff
@@ -52,7 +52,7 @@ class Splash extends React.Component {
 
         <HugeSpacer/>
 
-        {this.props.user && <UserGroups user={this.props.user}/>}
+        {this.props.user && this.props.user.visibleGroups && <UserGroups user={this.props.user}/>}
```

Comment:
I'm a little confused here how/why `this.props.user` was defined when there was no logged-in user.

Reply:
`user` is getting passed to the Splash component via the UserDataWrapper component, which by default is undefined.

Reply:
Ah, so looking closer, what's happening is that I have an existing login session (like most of our users will) but it lacks the `visibleGroups` attribute because that's only added by your previous PR. In order to get the new attribute with the current code, a person has to logout and log back in again. We can either guard against the missing attribute everywhere (a little ugly), or could put the computation somewhere else so it happens every time, not only in the login flow.

Reply:
Ah, right. @jameshadfield initially [pointed that out](https://github.com/nextstrain/nextstrain.org/pull/71#discussion_r363134923) as a consequence of moving the visibleGroups logic inside the setup of `OAuth2Strategy`.

I now recall that was the error I encountered in testing and fixed by logging in and out again.

Reply:
Is there no third solution where, now that the splash page is no longer broken, we can tell logged-in users to log in and out again? :smile:

Reply:
> edit: I realize now that we will not want to make users log in and out every time they get added to a new group. So this is not a real solution.

The Cognito groups for a user are (as I understand the code) only fetched when a user logs in, so I don't think it's possible (as currently implemented) to see a new cognito group without logging in again

Reply:
> The Cognito groups for a user are (as I understand the code) only fetched when a user logs in, so I don't think it's possible (as currently implemented) to see a new cognito group without logging in again

Correct. We can improve this in the future by refreshing user details from Cognito on ~every request (or, ideally, with some short-lived cache).